lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1004,105 +1011,103 @@ export class UserAgentApplication {
   acquireTokenSilent(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<string> {
     return new Promise<string>((resolve, reject) => {
       // Validate scopes
-      // TODO: Replace with new validation pattern
-      const isValidScope = this.validateInputScope(scopes);
-      if (isValidScope && !Utils.isEmpty(isValidScope)) {
+      try {
+        scopes = this.validateAndFilterInputScope(scopes);
+      } catch (err) {
+        if (err instanceof ClientConfigurationError) {
+          // These will return errors instead of strings in the future
+          // Should we throw instead of reject here?
+          reject(err.message);
+        } else {
+          reject(ErrorCodes.inputScopesError + Constants.resourceDelimeter + ErrorDescription.inputScopesError);
+        }
+      }
+
+      const scope = scopes.join(" ").toLowerCase();
```

Comment:
pretty sure this is either factored into a util or should be I have seen it all over

Reply:
I will do this as part of a separate PR.